<a href="https://colab.research.google.com/github/Nurhikmah05-alt/Data-Wrangling-Project1/blob/main/Tugas1_DataWrangling.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 1. Mengunggah Dataset ke Google Colab

In [ ]:
from google.colab import files
uploaded = files.upload()

Saving StudentPerformanceFactors.csv to StudentPerformanceFactors.csv


# 2. Membaca Dataset di Google Colab

In [ ]:
import pandas as pd
df = pd.read_csv("/content/StudentPerformanceFactors.csv")
df

,Hours_Studied,Attendance,Parental_Involvement,Access_to_Resources,Extracurricular_Activities,Sleep_Hours,Previous_Scores,Motivation_Level,Internet_Access,Tutoring_Sessions,Family_Income,Teacher_Quality,School_Type,Peer_Influence,Physical_Activity,Learning_Disabilities,Parental_Education_Level,Distance_from_Home,Gender,Exam_Score
0,23,84,Low,High,No,7,73,Low,Yes,0,Low,Medium,Public,Positive,3,No,High School,Near,Male,67
1,19,64,Low,Medium,No,8,59,Low,Yes,2,Medium,Medium,Public,Negative,4,No,College,Moderate,Female,61
2,24,98,Medium,Medium,Yes,7,91,Medium,Yes,2,Medium,Medium,Public,Neutral,4,No,Postgraduate,Near,Male,74
3,29,89,Low,Medium,Yes,8,98,Medium,Yes,1,Medium,Medium,Public,Negative,4,No,High School,Moderate,Male,71
4,19,92,Medium,Medium,Yes,6,65,Medium,Yes,3,Medium,High,Public,Neutral,4,No,College,Near,Female,70
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
6602,25,69,High,Medium,No,7,76,Medium,Yes,1,High,Medium,Public,Positive,2,No,High School,Near,Female,68
6603,23,76,High,Medium,No,8,81,Medium,Yes,3,Low,High,Public,Positive,2,No,High School,Near,Female,69
6604,20,90,Medium,Low,Yes,6,65,Low,Yes,3,Low,Medium,Public,Negative,2,No,Postgraduate,Near,Female,68
6605,10,86,High,High,Yes,6,91,High,Yes,2,Low,Medium,Private,Positive,3,No,High School,Far,Female,68


# 3. Menampilkan Dataset Menggunakan head(), info(), dan shape()

Untuk memahami struktur awal dataset dilakukan eksplorasi dasar sebagai berikut:

In [ ]:
df.head()

,Hours_Studied,Attendance,Parental_Involvement,Access_to_Resources,Extracurricular_Activities,Sleep_Hours,Previous_Scores,Motivation_Level,Internet_Access,Tutoring_Sessions,Family_Income,Teacher_Quality,School_Type,Peer_Influence,Physical_Activity,Learning_Disabilities,Parental_Education_Level,Distance_from_Home,Gender,Exam_Score
0,23,84,Low,High,No,7,73,Low,Yes,0,Low,Medium,Public,Positive,3,No,High School,Near,Male,67
1,19,64,Low,Medium,No,8,59,Low,Yes,2,Medium,Medium,Public,Negative,4,No,College,Moderate,Female,61
2,24,98,Medium,Medium,Yes,7,91,Medium,Yes,2,Medium,Medium,Public,Neutral,4,No,Postgraduate,Near,Male,74
3,29,89,Low,Medium,Yes,8,98,Medium,Yes,1,Medium,Medium,Public,Negative,4,No,High School,Moderate,Male,71
4,19,92,Medium,Medium,Yes,6,65,Medium,Yes,3,Medium,High,Public,Neutral,4,No,College,Near,Female,70


Fungsi `df.head` untuk menampilkan 5 baris pertama dataset untuk melihat contoh isi data dan struktur kolom.

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6607 entries, 0 to 6606
Data columns (total 20 columns):
 #   Column                      Non-Null Count  Dtype 
---  ------                      --------------  ----- 
 0   Hours_Studied               6607 non-null   int64 
 1   Attendance                  6607 non-null   int64 
 2   Parental_Involvement        6607 non-null   object
 3   Access_to_Resources         6607 non-null   object
 4   Extracurricular_Activities  6607 non-null   object
 5   Sleep_Hours                 6607 non-null   int64 
 6   Previous_Scores             6607 non-null   int64 
 7   Motivation_Level            6607 non-null   object
 8   Internet_Access             6607 non-null   object
 9   Tutoring_Sessions           6607 non-null   int64 
 10  Family_Income               6607 non-null   object
 11  Teacher_Quality             6529 non-null   object
 12  School_Type                 6607 non-null   object
 13  Peer_Influence              6607 non-null   obje

Fungsi `df.info()` untuk memberikan informasi lengkap mengenai: Jumlah entri, Jumlah kolom, Tipe data setiap kolom, Jumlah nilai non-null, dan Penggunaan memori.

In [ ]:
df.shape

(6607, 20)

Fungsi `df.shap`e menunjukkan dimensi dataset dalam bentuk tuple (baris, kolom).

# 4. Penjelasan Struktur Dataset

Berdasarkan hasil eksplorasi menggunakan info() dan shape(), diperoleh informasi sebagai berikut:

1.   Jumlah baris (observasi): 6607
2.   Jumlah kolom (variabel): 20
3.   Tipe datanya terdapat 7 kolom bertipe int64 (numerik) dan 13 kolom bertipe object (kategorikal)
4.   Ukuran memori: sekitar 1.0 MB

Beberapa kolom memiliki missing values, seperti:
1.   Teacher_Quality
2.   Parental_Education_Level
3.   Distance_from_Home

Variabel numerik meliputi:
1.   Hours_Studied
2.   Attendance
3.   Sleep_Hours
4.   Previous_Scores
5.   Tutoring_Sessions
6.   Physical_Activity
7.   Exam_Score

Variabel kategorikal meliputi:
1.   Parental_Involvement
2.   Access_to_Resources
3.   Extracurricular_Activities
4.   Motivation_Level
5.   Internet_Access
6.   Family_Income
7.   Teacher_Quality
8.   School_Type
9.   Peer_Influence
10.  Learning_Disabilities
11.  Parental_Education_Level
12.  Distance_from_Home
13.  Gender


Dataset ini dapat digunakan untuk analisis hubungan antara faktor internal dan eksternal terhadap performa akademik siswa.

# 5. Pembuatan Struktur Data JSON Secara Manual

In [ ]:
import json
df = [
    {
        "Movie_ID": "M001",
        "Judul": "Harry Potter and the Sorcerer's Stone",
        "Tahun_Rilis": 2001,
        "Genre": "Fantasy/Adventure",
        "Durasi_Menit": 152,
        "Rating_IMDB": 7.7,
        "Jumlah_Vote": 10936000,
        "Budget_Juta_USD": 125,
        "Pendapatan_Juta_USD": 974.8,
        "Negara": "USA/UK",
        "Bahasa": "English"
    },
    {
        "Movie_ID": "M002",
        "Judul": "Doctor Strange in the Multiverse of Madness",
        "Tahun_Rilis": 2022,
        "Genre": "Fantasy/Action",
        "Durasi_Menit": 126,
        "Rating_IMDB": 6.9,
        "Jumlah_Vote": 1053500,
        "Budget_Juta_USD": 414.9,
        "Pendapatan_Juta_USD": 955.8,
        "Negara": "USA",
        "Bahasa": "English"
    },
    {
        "Movie_ID": "M003",
        "Judul": "Fantastic Beasts and Where to Find Them",
        "Tahun_Rilis": 2016,
        "Genre": "Fantasy/Adventure",
        "Durasi_Menit": 132,
        "Rating_IMDB": 7.2,
        "Jumlah_Vote": 480000,
        "Budget_Juta_USD": 180,
        "Pendapatan_Juta_USD": 816.0,
        "Negara": "USA/UK",
        "Bahasa": "English"
    },
    {
        "Movie_ID": "M004",
        "Judul": "The Lord of the Rings: The Return of the King",
        "Tahun_Rilis": 2003,
        "Genre": "Fantasy/Adventure",
        "Durasi_Menit": 201,
        "Rating_IMDB": 9.0,
        "Jumlah_Vote": 1900000,
        "Budget_Juta_USD": 94,
        "Pendapatan_Juta_USD": 1140.6,
        "Negara": "New Zealand/USA",
        "Bahasa": "English"
    },
    {
        "Movie_ID": "M005",
        "Judul": "Oz the Great and Powerful",
        "Tahun_Rilis": 2013,
        "Genre": "Fantasy/Adventure",
        "Durasi_Menit": 130,
        "Rating_IMDB": 6.3,
        "Jumlah_Vote": 220000,
        "Budget_Juta_USD": 215,
        "Pendapatan_Juta_USD": 493.3,
        "Negara": "USA",
        "Bahasa": "English"
    },
    {
        "Movie_ID": "M006",
        "Judul": "The Sorcerer's Apprentice",
        "Tahun_Rilis": 2010,
        "Genre": "Fantasy/Adventure",
        "Durasi_Menit": 109,
        "Rating_IMDB": 6.1,
        "Jumlah_Vote": 180000,
        "Budget_Juta_USD": 150,
        "Pendapatan_Juta_USD": 215.3,
        "Negara": "USA",
        "Bahasa": "English"
    }
]

with open("dataset_film_magic.json", "w") as file:
    json.dump(df, file, indent=4)
df

[{'Movie_ID': 'M001',
  'Judul': "Harry Potter and the Sorcerer's Stone",
  'Tahun_Rilis': 2001,
  'Genre': 'Fantasy/Adventure',
  'Durasi_Menit': 152,
  'Rating_IMDB': 7.7,
  'Jumlah_Vote': 10936000,
  'Budget_Juta_USD': 125,
  'Pendapatan_Juta_USD': 974.8,
  'Negara': 'USA/UK',
  'Bahasa': 'English'},
 {'Movie_ID': 'M002',
  'Judul': 'Doctor Strange in the Multiverse of Madness',
  'Tahun_Rilis': 2022,
  'Genre': 'Fantasy/Action',
  'Durasi_Menit': 126,
  'Rating_IMDB': 6.9,
  'Jumlah_Vote': 1053500,
  'Budget_Juta_USD': 414.9,
  'Pendapatan_Juta_USD': 955.8,
  'Negara': 'USA',
  'Bahasa': 'English'},
 {'Movie_ID': 'M003',
  'Judul': 'Fantastic Beasts and Where to Find Them',
  'Tahun_Rilis': 2016,
  'Genre': 'Fantasy/Adventure',
  'Durasi_Menit': 132,
  'Rating_IMDB': 7.2,
  'Jumlah_Vote': 480000,
  'Budget_Juta_USD': 180,
  'Pendapatan_Juta_USD': 816.0,
  'Negara': 'USA/UK',
  'Bahasa': 'English'},
 {'Movie_ID': 'M004',
  'Judul': 'The Lord of the Rings: The Return of the King',
  

In [ ]:
df = pd.read_json("dataset_film_magic.json")
df.head()

,Movie_ID,Judul,Tahun_Rilis,Genre,Durasi_Menit,Rating_IMDB,Jumlah_Vote,Budget_Juta_USD,Pendapatan_Juta_USD,Negara,Bahasa
0,M001,Harry Potter and the Sorcerer's Stone,2001,Fantasy/Adventure,152,7.7,10936000,125.0,974.8,USA/UK,English
1,M002,Doctor Strange in the Multiverse of Madness,2022,Fantasy/Action,126,6.9,1053500,414.9,955.8,USA,English
2,M003,Fantastic Beasts and Where to Find Them,2016,Fantasy/Adventure,132,7.2,480000,180.0,816.0,USA/UK,English
3,M004,The Lord of the Rings: The Return of the King,2003,Fantasy/Adventure,201,9.0,1900000,94.0,1140.6,New Zealand/USA,English
4,M005,Oz the Great and Powerful,2013,Fantasy/Adventure,130,6.3,220000,215.0,493.3,USA,English


Mengonversi data dari JSON ke CSV

In [ ]:
df.to_csv("dataset_film_magic.csv", index=False)
df

,Movie_ID,Judul,Tahun_Rilis,Genre,Durasi_Menit,Rating_IMDB,Jumlah_Vote,Budget_Juta_USD,Pendapatan_Juta_USD,Negara,Bahasa
0,M001,Harry Potter and the Sorcerer's Stone,2001,Fantasy/Adventure,152,7.7,10936000,125.0,974.8,USA/UK,English
1,M002,Doctor Strange in the Multiverse of Madness,2022,Fantasy/Action,126,6.9,1053500,414.9,955.8,USA,English
2,M003,Fantastic Beasts and Where to Find Them,2016,Fantasy/Adventure,132,7.2,480000,180.0,816.0,USA/UK,English
3,M004,The Lord of the Rings: The Return of the King,2003,Fantasy/Adventure,201,9.0,1900000,94.0,1140.6,New Zealand/USA,English
4,M005,Oz the Great and Powerful,2013,Fantasy/Adventure,130,6.3,220000,215.0,493.3,USA,English
5,M006,The Sorcerer's Apprentice,2010,Fantasy/Adventure,109,6.1,180000,150.0,215.3,USA,English
